apis/cluster/v1alpha1/types.go

```diff
@@ -0,0 +1,222 @@
+/*
+Copyright 2021 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+)
+
+// ClusterParameters define the desired state of an ArgoCD Cluster
+type ClusterParameters struct {
+	// Server is the API server URL of the Kubernetes cluster
+	Server string `json:"server"`
+	// Name of the cluster. If omitted, will use the server address
+	Name string `json:"name"`
+	// Config holds cluster information for connecting to a cluster
+	Config ClusterConfig `json:"config"`
+	// Holds list of namespaces which are accessible in that cluster. Cluster level resources will be ignored if namespace list is not empty.
+	// +optional
+	Namespaces []string `json:"namespaces,omitempty"`
+	// RefreshRequestedAt holds time when cluster cache refresh has been requested
+	// TODO can we remove this field?
+	// +optional
+	RefreshRequestedAt *metav1.Time `json:"refreshRequestedAt,omitempty"`
```

Comment:
I feel like wee don't need this field in the managed resource. What do you think?